dags/pull_weather.py

```diff
@@ -0,0 +1,92 @@
+Then, 
+A Slack notification is raised when the airflow process fails.
+"""
+import sys
+from airflow import DAG
+from datetime import datetime, timedelta
+from airflow.operators.python_operator import PythonOperator
+from airflow.hooks.base_hook import BaseHook
+from airflow.contrib.operators.slack_webhook_operator import SlackWebhookOperator
+from airflow.hooks.postgres_hook import PostgresHook
+from airflow.models import Variable
+from dags.pull_wys import SLACK_CONN_ID 
+
+#connection credentials
+ec2_cred = PostgresHook("weather_bot")
+
+#import python scripts
+try:
+    sys.path.append('script-path')
+    from prediction_import import prediction_upsert
+    from mrc_import import historical_upsert
+except:
+    raise ImportError("script import failed")
+
+
+SLACK_CONN_ID = 'slack_data_pipeline'
+dag_config = Variable.get('slack_member_id', deserialize_json=True)
+list_names = dag_config['raphael'] + ' ' + dag_config['islam'] + ' ' + dag_config['natalie'] 
+
+
+def task_fail_slack_alert(context):
+    slack_webhook_token = BaseHook.get_connection(SLACK_CONN_ID).password
+
+    if context.get('task_instance').task_id == 'pull_prediction':
+        task_msg = """:cat_shock: The Task {task} in Pull Weather dag failed, 
+			{slack_name} please check.""".format(
+            task=context.get('task_instance').task_id, slack_name = list_names,)
+    
+    else:
+        task_msg = """ :eyes: The Task {task} in Pull Weather dag failed, 
+			{slack_name} please check.""".format(
+            task=context.get('task_instance').task_id, slack_name = list_names,)    
+        
+    # this adds the error log url at the end of the msg
+    slack_msg = task_msg + """ (<{log_url}|log>)""".format(
+            log_url=context.get('task_instance').log_url,)
+    failed_alert = SlackWebhookOperator(
+        task_id='slack_test',
+        http_conn_id='slack',
+        webhook_token=slack_webhook_token,
+        message=slack_msg,
+        username='airflow',
+        )
+    return failed_alert.execute(context=context)
+
+
+
+#DAG
+ 
+default_args = {
+    'owner':'rdumas',
+    'depends_on_past':False,
+    'start_date': datetime(2022, 11, 8),
+    'email': ['raphael.dumas@toronto.ca'],
+    'email_on_failure': False,
+    'email_on_success': False,
+    'retries': 0,
+    'retry_delay': timedelta(minutes=5),
+    'on_failure_callback': task_fail_slack_alert
+}
+
+dag = DAG('pull_weather', default_args=default_args, schedule_interval='@daily', catchup=False)
+
+#=======================================#
+#dag tasks
+
+with ec2_cred.get_conn() as conn:
+    PULL_PREDICTION = PythonOperator(
+        task_id = 'pull_prediction',
+        python_callable = prediction_upsert,
+        dag=dag,
+        op_args=[conn]
+    )
+
+    PULL_HISTORICAL = PythonOperator(
+        task_id = 'pull_historical',
+        python_callable = historical_upsert,
+        dag=dag,
+        op_args=[conn, {{ ds }}]
+    )
```

Comment:
Just noticed that we need to prevent backfilling the prediction puller task (it can only run for today to get the following days predictions). This [post](https://www.mikulskibartosz.name/latestonlyoperator-skip-airflow-task-while-running-backfill/) shows how to use `LatestOnlyOperator` to skip backfilling some tasks. So, our code needs something like the following:
```
from airflow.operators.latest_only_operator import LatestOnlyOperator

no_backfilling_allowed = LatestOnlyOperator(task_id="no_backfilling_allowed", dag=dag)

PULL_HISTORICAL
no_backfilling_allowed >> PULL_PREDICTION
```

Reply:
This temporarily applies for the pull_historical data as well, I'll also set no_backfill for the historical data until it is upgraded